### PR TITLE
[ISSUE 21] Adding timezone support

### PR DESCRIPTION
**Pull request by soojinj**
Issue #, if available:
#21

Description of changes:
SDK side of changes to support timezone

v2 of https://github.com/awslabs/aws-athena-query-federation/pull/130

### CODE REVIEW

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java [hunk 1]
```diff
@@ -196,20 +200,54 @@ else if (vector instanceof StructVector) {
...
+             * @param o1 object with unsupported type
+             * @return another object with supported/converted type
+             */
+
+            Object value = rawValue;
+            if (rawValue instanceof org.joda.time.LocalDateTime) {
+                DateTimeZone dtz = ((org.joda.time.LocalDateTime) rawValue).getChronology().getZone();
+                long dateTimeWithTimeZone = ((org.joda.time.LocalDateTime) rawValue).toDateTime(dtz).getMillis();
+                value = DateTimeFormatterUtil.constructZonedDateTime(dateTimeWithTimeZone);
+            }
```
Comment (avirtuos) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java:227]: Why do this for every single value? This check is irrelevant for 9 out of 10 types yet we'd take the branching perf hit for all of them. Can this be moved into the relevant switch statement path?

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/MetadataHandler.java [hunk 2]
```diff
@@ -354,6 +354,7 @@ public GetTableLayoutResponse doGetTableLayout(final BlockAllocator allocator, f
         ) {
             Block partitions = allocator.createBlock(partitionSchemaBuilder.build());
             partitions.constrain(constraintEvaluator);
+            partitions.constrain(constraintEvaluator);
```
Comment (avirtuos) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/MetadataHandler.java:357]: Why adding the duplicate line / call?

**Comment by avirtuos** [comment]
Ok, looks like only remaining things from my perspective are:

1. Please add new documentation about this type to the federation-sdk readme
2. Do not push this until the corresponding Athena side changes are shipped.

**Comment by soojinj** [comment]
testing:
select isodatetimeest1, isodatetimeest1 at time zone 'UTC' from sometable;
        isodatetimeest1         |            _col1            
--------------------------------+-----------------------------
 2015-12-21 17:42:34.000 UTC    | 2015-12-21 17:42:34.000 UTC 
 2015-12-21 17:42:34.000 -03:00 | 2015-12-21 20:42:34.000 UTC 
 2015-12-21 17:42:34.000 -05:00 | 2015-12-21 22:42:34.000 UTC 


select isodatetimeest1, isodatetimeest1 at time zone 'UTC' from sometable WHERE isodatetimeest1 < timestamp '2015-12-21 18:42:34 UTC';
       isodatetimeest1       |            _col1            
-----------------------------+-----------------------------
 2015-12-21 17:42:34.000 UTC | 2015-12-21 17:42:34.000 UTC 
(1 row)

**Comment by avirtuos** [comment]
Nice job @soojinj , thanks for powering through this.

**Pull request merged**